Tasks/VsTestV2/inputparser.ts

```diff
@@ -550,8 +553,16 @@ function getTestPlatformPath(inputDataContract : idc.InputDataContract) {
 
     const vsVersion: number = parseFloat(vsTestVersion);
 
+    if (vsVersion === 17.0) {
+        const vstestconsolePath = getVSTestConsolePath('17.0', '18.0');
+        if (vstestconsolePath) {
+            return path.join(vstestconsolePath, 'Common7', 'IDE', 'Extensions', 'TestPlatform');
+        }
+        throw (new Error(tl.loc('VstestNotFound', utils.Helper.getVSVersion(vsVersion))));
+    }
+
     if (vsVersion === 16.0) {
-        const vstestconsolePath = getVSTestConsolePath('15.0', '17.0');
+        const vstestconsolePath = getVSTestConsolePath('16.0', '17.0');
```

Comment:
just for clarification, why are we removing 15?

Reply:
This is actually a bug, because, getVSTestConsolePath function will search for vsVersion in range of [15,17) and if user has both vs17 and vs19 installed on machine and if explicitly provides vs19 in execution then also task would resolve to vs17 since search happens in incremental manner. More details can be found here : https://github.com/microsoft/azure-pipelines-tasks/issues/15134